### PR TITLE
feat: Shows link location and respective icon in /bookings

### LINKED ISSUE #11730: [CAL-2590] Show link location in /bookings
show link to join a call in the /booking overview

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2590](https://linear.app/calcom/issue/CAL-2590/show-link-location-in-bookings)</sub>

### PR DESCRIPTION
**Pull request by siddharth-movaliya**
## What does this PR do?

Shows link location and respective icon in /bookings.

Fixes #11730 

/claim #11730

Screenshot 2023-10-12 at 11 26 06 PM

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

Once done a booking, go to the bookings page.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

**Review by PeerRich — DISMISSED** [review]
we should not hard code logos and names, instead use whatever conference location was used

**Comment by Siddharth-2382** [comment]
Okay understood....will make the changes

**Comment by Siddharth-2382** [comment]
View of bookings page after the latest changes:

https://github.com/calcom/cal.com/assets/94699055/590efd0e-655d-4a7b-8495-fd4d11ae5833

File: apps/web/components/booking/BookingListItem.tsx [hunk 1]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
...
+                <Link href={bookingLink} className="text-sm leading-6 text-blue-400 hover:underline">
+                  <div className="flex items-center gap-2">
+                    {eventLocationType ? (
+                      <>
+                        <img
+                          src={eventLocationType.iconUrl}
+                          className="h-4 w-4 rounded-sm"
+                          alt={`${eventLocationType.label} logo`}
+                        />
+                        Join {eventLocationType.label}
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:367]: can you add this to en/common.json and use `t("join", eventLocationType.label)` ?

File: apps/web/components/booking/BookingListItem.tsx [hunk 2]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
...
+                      </>
+                    ) : (
+                      <>Join Meeting</>
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:370]: same here, t("join_meeting")

File: apps/web/components/booking/BookingListItem.tsx [hunk 3]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
                   attendees={booking.attendees}
                 />
               </div>
+              {booking.location && (
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:357]:
```suggestion
              {!isPending && booking.location && (
```

File: apps/web/components/booking/BookingListItem.tsx [hunk 4]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
...
+                      <>
+                        <img
+                          src={eventLocationType.iconUrl}
+                          className="h-4 w-4 rounded-sm"
+                          alt={`${eventLocationType.label} logo`}
+                        />
+                        Join {eventLocationType.label}
+                      </>
+                    ) : (
+                      <>Join Meeting</>
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:370]:
```suggestion
                      <>t("join_meeting")</>
```

File: apps/web/components/booking/BookingListItem.tsx [hunk 5]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
...
+                <Link href={bookingLink} className="text-sm leading-6 text-blue-400 hover:underline">
+                  <div className="flex items-center gap-2">
+                    {eventLocationType ? (
+                      <>
+                        <img
+                          src={eventLocationType.iconUrl}
+                          className="h-4 w-4 rounded-sm"
+                          alt={`${eventLocationType.label} logo`}
+                        />
+                        Join {eventLocationType.label}
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:367]:
```suggestion
                        t("join", eventLocationType.label)
```

Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:367]:
```suggestion
                        {t("join_event_location", { eventLocationType: eventLocationType.label })}
```

File: apps/web/components/booking/BookingListItem.tsx [hunk 6]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
...
+                      <>
+                        <img
+                          src={eventLocationType.iconUrl}
+                          className="h-4 w-4 rounded-sm"
+                          alt={`${eventLocationType.label} logo`}
+                        />
+                        {t("join_event_location", { eventLocationType: eventLocationType.label })}
+                      </>
+                    ) : (
+                      <>t("join_meeting")</>
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:370]:
```suggestion
                      t("join_meeting")
```

**Comment by Siddharth-2382** [comment]
@PeerRich sorry for not responding. There were some exams going on so.

File: apps/web/components/booking/BookingListItem.tsx [hunk 7]
```diff
@@ -353,6 +354,25 @@ function BookingListItem(booking: BookingItemProps) {
                   attendees={booking.attendees}
                 />
               </div>
+              {!isPending && booking.location && (
+                <Link href={bookingLink} className="text-sm leading-6 text-blue-400 hover:underline">
```
Comment (PeerRich) [inline apps/web/components/booking/BookingListItem.tsx:358]: the booking link is not the video link.

clicking the "join" button opens the booking details, not the location URL

Comment (sean-brydon) [reply apps/web/components/booking/BookingListItem.tsx:358]: I think we can get the video URL by doing something as follows in the booking get handler

https://github.com/calcom/cal.com/pull/11942/commits/cecd2ae537d0e601df7a8c9d6deb3b3719792e32

(just a git diff for reference) 

you should be able to access booking.metadata.videoUrl (I havent tested but the data is there in the DB :P)

Comment (Siddharth-2382) [reply apps/web/components/booking/BookingListItem.tsx:358]: working on it rn.

Comment (Siddharth-2382) [reply apps/web/components/booking/BookingListItem.tsx:358]: Seems like the booking prop which is being used on this page isn't having a metadata.
Screenshot 2023-10-17 at 6 10 51 PM

Comment (Siddharth-2382) [reply apps/web/components/booking/BookingListItem.tsx:358]: tried booking.refrences[0].meetingUrl, it's giving meeting links for google meet and zoom video but not for cal video or a event that has external meeting link.

**Review by PeerRich — DISMISSED** [review]
link to video meeting is missing

File: apps/web/components/booking/BookingListItem.tsx (hunk 7 above)
Comment (sean-brydon) [reply apps/web/components/booking/BookingListItem.tsx:358]: Hmmm - this appears to be along the right lines https://github.com/calcom/cal.com/blob/75d7d2f17291e7b1e137839b1de3eac998ecfa0d/apps/web/pages/booking/%5Buid%5D.tsx#L117 on how to get this info. Looks like its a bit complicated to get but deffo possible if you follow it through. 

There is also this fn which might help https://github.com/calcom/cal.com/blob/75d7d2f17291e7b1e137839b1de3eac998ecfa0d/packages/app-store/locations.ts#L392

Comment (Siddharth-2382) [reply apps/web/components/booking/BookingListItem.tsx:358]: yes I was currently doing that. Just need more time.

Comment (Siddharth-2382) [reply apps/web/components/booking/BookingListItem.tsx]: @sean-brydon thanks for the help. Seems like I was just being dumb and I thought cal video was a default app but then I found daily.co api integration. The meeting links now work fine for both the apps one and external meeting link types.

**Comment by Siddharth-2382** [comment]
@PeerRich I got it! ;)

https://github.com/calcom/cal.com/assets/94699055/51110c12-4f0c-4c64-a5e2-5a5b664858aa

**Review by sean-brydon — DISMISSED** [review]
Overall this is great! Few nits that i think should be addressed

You can't click the link -> You have to middle click / right click open with tab (due to event propagation it seems)

Bookings with "link" as their location and also the likes of whereby just show empty with no location 
CleanShot 2023-10-18 at 11 31 29@2x

**Comment by Siddharth-2382** [comment]
> Overall this is great! Few nits that i think should be addressed
> 
> You can't click the link -> You have to middle click / right click open with tab (due to event propagation it seems)
> 
> Bookings with "link" as their location and also the likes of whereby just show empty with no location CleanShot 2023-10-18 at 11 31 29@2x

Can you review again with the latest changes? I think I fixed that problem in the latest commits.

**Comment by sean-brydon** [comment]
> > Overall this is great! Few nits that i think should be addressed
> > You can't click the link -> You have to middle click / right click open with tab (due to event propagation it seems)
> > Bookings with "link" as their location and also the likes of whereby just show empty with no location CleanShot 2023-10-18 at 11 31 29@2x
> 
> Can you review again with the latest changes? I think I fixed that problem in the latest commits.

Works perfect

**Comment by sean-brydon** [comment]
Once type check is address im happy to merge this great work

**Comment by Siddharth-2382** [comment]
> Once type check is address im happy to merge this great work

done :)

**Review by PeerRich — APPROVED** [review]

**Comment by kremedev** [comment]
> @PeerRich I got it! ;)
> 
>  demo.mov

Why second booking link say "Join Meeting" when it redirects to Google Meet?

**Comment by Siddharth-2382** [comment]
Because it's not a Google meet app meeting it's an external link meet which means it is supposed to show join meeting.

**Comment by kremedev** [comment]
FaceTime, WhatsApp, and Discord links are supposed to show "Join Meeting" too?

**Comment by Siddharth-2382** [comment]
> FaceTime, WhatsApp, and Discord links are supposed to show "Join Meeting" too?

If the meeting location is set to FaceTime, WhatsApp, Discord or any other app, it will show the respective app icon and say `Join {APP NAME}`. If the location of the meet is `Link Meeting`, it will just show the message `Join Meeting`. I hope you understood this.

**Comment by kremedev** [comment]
It shows "Join Meeting" when meeting locations are FaceTime, WhatsApp, and Discord.

https://github.com/calcom/cal.com/assets/134702704/76225bf5-666e-4767-b3a2-270147ad3f6d

**Comment by Siddharth-2382** [comment]
Those apps while having the meeting location of a particular app still needs to be given an external link and hence it shows Join Meeting as it is more of a general approach for any meetings that were given an external link and not just created by cal.

**Pull request merged**